app/static/js/userProfile.js

```diff
@@ -1,323 +1,367 @@
+        success: function(response) {
+            console.log(response);
+            // Handle success response if needed
+        },
+        error: function(error) {
+            console.error("An error occurred:", error);
+            // Handle error if needed
+        }
+    });
+  });
+
+
+  function changeAction(e){
+    let profileAction = $(this).val()
+    let username = $(this).data('username')
+    if (profileAction == "Emergency Contact"){
+      window.location.href = `/profile/${username}/emergencyContact`
+    } else if (profileAction == "Insurance Information"){
+      window.location.href = `/profile/${username}/insuranceInfo`
+    } else if(profileAction == "Print Travel Form"){
+      printDocument(`/profile/${username}/travelForm`)
+    } else if (profileAction == "View Service Transcript"){
+      window.location.href = `/profile/${username}/serviceTranscript`
+    } else if (profileAction == "Manage CCE Minor") {
+      window.location.href = `/profile/${username}/cceMinor`
+    }
+    $(this).val('')
+  }
+
+  // This function is to disable all the dates before current date in the ban modal End Date picker
+  $(function(){
+    var banEndDatepicker = $("#banEndDatepicker");
+    banEndDatepicker.datepicker({
+      changeYear: true,
+      changeMonth: true,
+      minDate:+1,
+      dateFormat: "yy-mm-dd",
+    }).attr('readonly','readonly');
+  });
+
+    /*
+     * Ban Functionality
+     */
+
+
+  var programID;
+  $(".ban").click(function() {
+    $.ajax({
+      url: `/profile/${$(this).data("username")}/removeFromTranscript/${$(this).data("programid")}`,
+      type: "GET",
+      success: function(response) {
+          // Check if the program is marked for removal from transcript
+          if (response.removedFromTranscript) {
+              // Handle the case where program is marked for removal
+              $('.removeFromTranscriptCheckbox').prop('checked', true);
+          } else {
+              // Handle the case where program is not marked for removal
+              $('.removeFromTranscriptCheckbox').prop('checked', false);
+          }
+      },
+      error: function(error, status) {
+          console.log(error, status);
+      }
+    });
+  
+    var banButton = $("#banButton")
+    var banEndDateDiv = $("#banEndDate") // Div containing the datepicker in the ban modal
+    var banEndDatepicker = $("#banEndDatepicker") // Datepicker in the ban modal
+    var banNoteDiv = $("#banNoteDiv") // Div containing the note displaying why the user was banned previously
+                                     //Should only diplay when the modal is going to unban a user
+    var banNote = $("#banNote")
+
+    banButton.text($(this).val() + " Volunteer");
+    programID = $(this).data("programid"); // Assign value to programID variable
+    banButton.data("programID", $(this).data("programid"))
+    banButton.data("username", $("#notifyInput").data("username"))

```

Comment:
While this object does have the username of the user, it feels weird that you're using something which isn't really meant to be global to the page and that's not totally related to the ban model you're using it in. The whole thing is a refactor. It's odd we don't have a hidden input on the user's profile page with their username in it but we have 10 data-usernames. You could use the edit button which currently has the class '.ban' which you'll hopefully be changing soon since that's a little more related and still has a data-username attribute already attached to it.